test/functional/ui/bufhl_spec.lua

```diff
@@ -534,6 +534,44 @@ describe('Buffer highlighting', function()
     ]]}
   end)
 
+  it('respects priority', function()
+    local set_extmark = curbufmeths.set_extmark
+    local id = meths.create_namespace('')
+    insert [[foobar]]
+
+    set_extmark(id, 0, 0, {
+      end_line = 0,
+      end_col = 6,
+      hl_group = "Normal",
+      priority = 100
+    })
+    set_extmark(id, 0, 3, {
+      end_line = 0,
+      end_col = 6,
+      hl_group = "String",
+      priority = 1
+    })
+
+    screen:expect { unchanged = true }
```

Comment:
what's this testing?

Reply:
If the priority was not working, the `String` highlight would have shown up.
Adding  the `Normal` with higher priority should return the screen to that original state.

Reply:
we prefer tests that actually checks the redraw if possible. Here you could change the range of the `"Normal"` highlight to be smaller than (0,6), then we can confirm the difference in behaviour between the different parts.

Reply:
Adding a proper test helped : priority was not working at all because I am bad at writing C